### DISCUSSION TITLE
Followup on Issue #1067: Authorization need Admin Approval?

### DISCUSSION THREAD

**Discussion by AJR07** [abraunegg/onedrive#2745] [Q&A] [answered]
Following up on Issue #1067, I encountered the same issue when running `onedrive`, and then following the link provided. After signing in, I was met with the following error:
![image](https://github.com/abraunegg/onedrive/assets/65659586/6bde2cfc-d9d9-4f95-831b-49ee133420ec)

I followed the steps outlined by [abraunegg](https://github.com/abraunegg) in Issue #1067,
> This is due to that the client identifier was updated, along with the authorisation scopes to be compatible with the OneDrive API changes. Please refer to #835 for more details.

In Issue #835, I tried:
1. Stopping all onedrive services, logging out and then re-authenticating
2. Updating my configuration file to include `application_id = "22c49a0d-d21c-4792-aed1-8f163c982546"`

This is my configuration file:

```
Configuration file successfully loaded
onedrive version                             = v2.4.25
Config path                                  = /home/ajr/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /home/ajr/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = 
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 22c49a0d-d21c-4792-aed1-8f163c982546
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 600
Config option 'ip_protocol_version'          = 0
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

Is there a way to get around this so I can sync without admin approval? I am able to do so with Onedrive Sync on windows. Thanks!

**Comment by abraunegg** [comment, marked as answer]
@AJR07 
Do not use the skilion client identifier.

Your *only* resolution is to seek admin approval through your IT Admin that look after your Azure Active Directory.

**Comment by AJR07** [comment]
Okay thanks!

**Comment by doronbehar** [comment]
I have a question regarding this issue: `rclone` _is_ capable of accessing my onedrive files, when permitting it to access my onedrive I didn't encounter the above issue. Any idea how is that? Also, how can I guide my IT department to give the option to `abraunegg.github.io` to access users' onedrives? Is https://portal.azure.com the correct place to look? Help will be appreciated. Thanks :pray:.

**Reply by abraunegg** [reply]
@doronbehar 
> I have a question regarding this issue: rclone is capable of accessing my onedrive files, when permitting it to access my onedrive I didn't encounter the above issue. Any idea how is that?

You need to:
* Ask 'rclone' forums that question - not here.
* Ask your organisation why - not here. The most likely answer is that they approved it.

> Also, how can I guide my IT department to give the option to abraunegg.github.io to access users' onedrives? Is https://portal.azure.com/ the correct place to look? Help will be appreciated.

If your IT department do not know how to do this, the best avenue of guidance can be found here: https://github.com/abraunegg/onedrive/blob/master/docs/national-cloud-deployments.md

Every organisation is different - augment these steps to make it work for your organisation.

**Reply by doronbehar** [reply]
> You need to:
>
>   * Ask 'rclone' forums that question - not here.

Haha what stops them from replying the same thing? It is still worth a trial though, so I asked them [here](https://forum.rclone.org/t/question-about-onedrive-authentication-methods/49063).

> If your IT department do not know how to do this, the best avenue of guidance can be found here: [`master`/docs/national-cloud-deployments.md](https://github.com/abraunegg/onedrive/blob/master/docs/national-cloud-deployments.md)

Thanks for the link, it would have been nice if these documents would have been organized in a more searchable manner. Also, are you sure that the nationality is the only thing that makes these deployments get strict data security requirements? I'm not a resident of Germany, nor China, nor the US.. Maybe it would be nice to mention these instructions in [this paragraph](https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#authorise-the-application-with-your-microsoft-onedrive-account)?
